addons/mail/static/src/models/thread.js

```diff
@@ -431,8 +431,15 @@ registerModel({
                     limit,
                 },
             });
-            return this.insert(channelsData.map(
-                channelData => this.convertData(channelData)
+            return this.insert(channelsData.map(channelData =>
+                this.messaging.models['Thread'].convertData({
+                    channel: insertAndReplace({
+                        channel_type: channelData.channel_type,
+                        id: channelData.id,
+                    }),
+                    id: channelData.id,
+                    name: channelData.name,
+                })
```

Comment:
the `channel` key in data should contain everything, it shouldn't be necessary to manually list all fields here

Reply:
I did it this way because the method used here to retrieve the data is not `channel_info` but `search_read`, so there is no `channel` key in `channelData`.